site/_js/events.js

```diff
@@ -34,54 +143,115 @@ const getEvents = async () => {
     upcomingEvents: all.filter(event => !event.isPastEvent),
     pastEvents: all.filter(event => event.isPastEvent),
   };
-};
+});
 
-const getMemoizedEvents = memoize(getEvents);
-
-const eventLists = [
-  {
-    loadMore: document.getElementById('load-upcoming-events'),
-    container: document.getElementById('upcoming-events'),
-    fetchItems: async (skip, take) => {
-      const {upcomingEvents} = await getMemoizedEvents();
-
-      return upcomingEvents
-        .sort((a, b) => a.date.localeCompare(b.date))
-        .slice(skip, take + skip)
-        .map(event => event.html);
-    },
-  },
-  {
-    loadMore: document.getElementById('load-past-events'),
-    container: document.getElementById('past-events'),
-    fetchItems: async (skip, take) => {
-      const {pastEvents} = await getMemoizedEvents();
-
-      return pastEvents
-        .sort((a, b) => b.date.localeCompare(a.date))
-        .slice(skip, take + skip)
-        .map(event => event.html);
-    },
-  },
-];
-
-eventLists.forEach(list => {
-  if (list.loadMore === null || list.container === null) return;
-
-  const skip = list.container.querySelectorAll('enhanced-event-card').length;
-
-  loadMore(
-    list.loadMore,
-    list.container,
-    list.fetchItems,
-    () => {
-      document
-        .getElementById('loading-error')
-        ?.classList.remove('display-none');
-    },
-    {
-      skip,
-      take: 10,
+const filterEvents = events => {
+  return events.filter(event => {
+    if (
+      activeFilters.locations?.length &&
+      !activeFilters.locations.some(location => location === event.location)
+    ) {
+      return false;
     }
-  );
-});
+
+    if (
+      activeFilters.topics?.length &&
+      !activeFilters.topics.some(topic => event.topics.includes(topic))
+    ) {
+      return false;
+    }
+
+    if (
+      activeFilters.googlers?.length &&
+      !activeFilters.googlers.some(googler => event.googlers.includes(googler))
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+};
+
+function updateTagPills() {
+  if (!activeFiltersList) return;
+
+  activeFiltersList.items = Object.entries(activeFilters).flatMap(i => {
+    return i[1].map(value => ({key: i[0], value: value}));
+  });
+}
+
+/*
+Used exclusively to modify the margin when pills
+are present. Can be replaced with a :has selector
+once FF supports it.
+*/
+function updateFiltersAttribute() {
+  if (Object.values(activeFilters).some(i => i.length > 0)) {
+    wrapper?.setAttribute('data-active-filters', '');
+  } else {
+    wrapper?.removeAttribute('data-active-filters');
+  }
+}
+
+function addMobileListeners() {
+  /** @type {HTMLDialogElement|null} */
+  const filters = document.querySelector('#mobile-filters');
+  const opener = document.querySelector('#mobile-filters-opener');
+  const done = document.getElementById('mobile-filters-done');
+  const reset = document.getElementById('mobile-filters-reset');
+
+  // @ts-ignore
```

Comment:
Adding  /** @type {HTMLElement} */  to querySelector line should fix this ts issue, i think

Examples:
https://github.com/GoogleChrome/developer.chrome.com/blob/5b486c7e09faf1403c805d47896a50cbe93a6900/site/_js/web-components/navigation-tree.js#L35

https://github.com/GoogleChrome/developer.chrome.com/blob/5b486c7e09faf1403c805d47896a50cbe93a6900/site/_js/web-components/top-nav.js#L35

Reply:
Ah this relates to the typescript issue I mentioned a while back. The variable is already hinted as a HTMLDialogElement but our version of TS doesn't recognize the showModal method. The plan is to raise a separate issue which will link to these lines once merged